main.go

```diff
@@ -437,26 +474,78 @@ func createSecretCacheConfig(platform cluster.Platform) map[string]cache.Config
 			operatorNs = "redhat-ods-operator" // fall back case
 		}
 		namespaceConfigs[operatorNs] = cache.Config{}
-	case cluster.SelfManagedRhoai:
-		namespaceConfigs["redhat-ods-applications"] = cache.Config{}
+		return namespaceConfigs, nil
 	default:
-		namespaceConfigs["opendatahub"] = cache.Config{}
+		cNamespaceList := &corev1.NamespaceList{}
+		// if user create namespace and want it to be used as application namespace
+		// they need to create label "opendatahub.io/application-namespace": "true" then set DSCI to use it
+		// we only support one namespace in the cluster has this label
+		// if other namespace has generated-namespace label, we wont use such namespace
+		labelSelector := client.MatchingLabels{
+			labels.CustomizedAppNamespace: labels.True,
+		}
+		if err := cli.List(ctx, cNamespaceList, labelSelector); err != nil {
+			return map[string]cache.Config{}, err
+		}
+		switch len(cNamespaceList.Items) {
+		case 0:
+			if platform == cluster.SelfManagedRhoai {
```

Comment:
maybe we can have a method or a map that holds the mapping between the platform type and the default namespace name so we have a central place to look for and avoid duplication.

Reply:
wrap the code into a new function `getCommonCache()`